sql/table_cache.h

```diff
@@ -20,10 +20,10 @@
 
 struct Share_free_tables
 {
-  typedef I_P_List <TABLE, TABLE_share> List;
+  /** MY_ALIGNED - Avoid false sharing between instances */
+  typedef I_P_List <TABLE, TABLE_share> List
+    MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE);
```

Comment:
Again: "false sharing between instances". Appropriate target for alignment is ```Share_free_tables``` itself. OTOH ```Share_free_tables``` now contains only 1 element, so it can be optimised away.

You also need to make sure that ```lf_hash``` does aligned alloc. This might be complex, because for each element ```lf_hash``` uses first ```LF_HASH_OVERHEAD``` bytes for hash internal structures.